### PR TITLE
feat(ls): add partition builder utility for list output grouping

### PR DESCRIPTION
**Pull request by kecbigmt**
## Summary
- Add `buildPartitions()` utility that groups sorted items into display partitions for `mm ls`
- Support single placement, date range, and numeric range partitioning
- Generate section stubs, skip empty partitions, apply range limits (100), and warn on item-head events
- Include 21 unit tests covering all partition types and edge cases

This is the partition builder component of the 004_list spec (Workstream C), which will be integrated with the formatter (Workstream D) in the CLI command.

## Test plan
- [x] All 21 unit tests pass for partition builder
- [x] All existing tests pass (`deno task test`)
- [x] Lint and format checks pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/presentation/cli/partitioning/build_partitions.ts [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+
+  // Start from 'to' and work backwards (newest first)
+  let current = toDate;
+  while (current >= fromDate && dates.length < limit) {
+    const iso = current.toISOString().slice(0, 10);
+    const dayResult = parseCalendarDay(iso);
+    if (dayResult.type === "ok") {
+      dates.push(dayResult.value);
+    }
+    current = new Date(current.getTime() - 24 * 60 * 60 * 1000);
```
[the remark this answers is not in the document]
Comment (kecbigmt) [reply src/presentation/cli/partitioning/build_partitions.ts]: Fixed in commit 7ff8a76. I used `setUTCDate()` instead of `setDate()` to ensure consistent behavior regardless of the local timezone, since CalendarDay uses UTC internally.
